src/infrastructure/diConfig.ts

```diff
@@ -0,0 +1,53 @@
+import type { AwilixContainer, Resolver } from 'awilix'
+import { asClass, asFunction, Lifetime } from 'awilix'
+import type { FastifyInstance } from 'fastify'
+import type P from 'pino'
+
+import { FakeIntegrationApiClient } from '../integrations/fakeIntegration/client/FakeIntegrationApiClient'
+import { CacheService } from '../modules/cache/CacheService'
+
+import type { Config } from './config'
+import { getConfig } from './config'
+
+export type ExternalDependencies = {
+  app?: FastifyInstance
+  logger?: P.Logger
+}
+export const SINGLETON_CONFIG = { lifetime: Lifetime.SINGLETON }
+
+export type DependencyOverrides = Partial<DiConfig>
+
+export function registerDependencies(
```

Comment:
I had to introduce dependency injection mechanism in order to make code testable. hardcoded dependencies that are instantiated on import require very dirty hacks to become testable, and I preferred to go the cleaner route with proper dependency injection